### PR TITLE
fix: request recording in unittest setUp method

### LINKED ISSUE #332: request recordings generated when testing Django
Running the Django tests (which are based on `unittest`) for Django v4.1 generates request recordings in addition to recordings for the test cases. The agent is trying to [disable request recording](https://github.com/getappmap/appmap-python/blob/47d8cc32adfbce52d7f935dd68229eb7454e66c1/_appmap/testing_framework.py#L120), but this doesn't appear to be sufficient.

### PR DESCRIPTION
**Pull request by zermelo-wisen**
Fixes #332

- Hooks `unittest` `TestCase._callSetUp` method to disable request recording in order to prevent the creation of a request recording when a request is made inside `setUp` method of a `unittest` test.
- Adds a `unittest` test to `django` fixture project which makes a request inside `setUp` method. This test makes the `test_django.py::TestDjangoApp::test_enabled` pass with the fix and fail without the fix.
- This edge case can be observed in this test in `django` project:
   `$ APPMAP=TRUE ./runtests.py auth_tests.test_views.ChangelistTests.test_user_change_email`
A request made inside `ChangelistTests.setUp` method leads to the creation of requests recording AppMap without the fix.

### CODE REVIEW

File: _appmap/test/test_django.py [hunk 1]
```diff
@@ -213,7 +213,7 @@ def test_enabled(self, pytester):
     def test_disabled(self, pytester, monkeypatch):
         monkeypatch.setenv("_APPMAP", "false")
         result = pytester.runpytest("-svv", "-p", "no:randomly", "-k", "test_request")
-        result.assert_outcomes(passed=1, failed=0, errors=0)
+        result.assert_outcomes(passed=2, failed=0, errors=0)
```
Comment (apotterri) [inline _appmap/test/test_django.py:216]: This also needs to check that no request recordings were created.

**Review by apotterri — CHANGES_REQUESTED** [review]

File: _appmap/test/test_django.py (hunk 1 above)
Comment (zermelo-wisen) [reply _appmap/test/test_django.py:216]: Doesn't the following line assert that as well?

https://github.com/getappmap/appmap-python/blob/f90ca457a33bdb9cd4eb808b3da27e9accb3e2d4/_appmap/test/test_django.py#L217

Comment (apotterri) [reply _appmap/test/test_django.py:216]: Well, it does, in the case that `APPMAP=false` (i.e. recording is completely disabled). The test I asked for ensures that request recordings aren't generated when other AppMaps are.

**Review by apotterri — APPROVED** [review]
Looks good. Thanks!
